src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -2520,3 +2520,34 @@ def _aks_mesh_update(
         return None
 
     return aks_update_decorator.update_mc(mc)
+
+
+def start_chat(prompt=None):
+    from azext_aks_preview._openai_wrapper import setup_openai, \
+        prompt_chat_gpt, SYSTEM_PROMPT, getch, \
+        prompt_user_to_run_script
+    errors, params = setup_openai()
+    if errors:
+        for e in errors:
+            print(e)
+        return
+
+    print("Please enter your request below.")
+    print("For example: Create a AKS cluster")
+
+    scripts, messages = prompt_chat_gpt([SYSTEM_PROMPT], params, start_input=prompt)
+    while True:
+        print("\nMenu: [p: re-Prompt, ", end="")
+        if len(scripts) > 0:
+            print("r: Run, ", end="")
+        print("q: Quit]", flush=True)
+
+        # Handle user input
+        user_input = getch()
+        if user_input in ('p', 'P'):
+            scripts, messages = prompt_chat_gpt(messages, params, insist=False, scripts=scripts)
```

Comment:
Could you please help explain the purpose of passing `insist=False` for `prompt_char_gpt`?

Reply:
at the very beginning of the session, we need to start with a user mission/task 
"Prompt:  I need to create a aks cluster"

If user just press enter without entering anything, we have so objective, so insist is default to True to prompt the user until they supply to task to work with

for example, user might hit enter by mistake multiple times until typed "Create a AKS cluster", so I only show "Prompt:"
<img width="1205" alt="image" src="https://github.com/Azure/azure-cli-extensions/assets/24964493/3aa388b6-17e0-45dd-a9ed-73e341320021">

Once that is started, user can always re-prompt the LLM to refine their objectives or add more requirements by re-prompting.  But they would trigger that by mistake or they change their mind,  so insist=False allow the user to not enter anything (= cancel out of the prompt)

<img width="664" alt="image" src="https://github.com/Azure/azure-cli-extensions/assets/24964493/eecae9b1-88e8-48c7-a01d-201ca972ec8c">